### PR TITLE
Migrate `@woocommerce/csv-export` to TS

### LINKED ISSUE #33102: [Enhancement]: Migrate `@woocommerce/csv-export` to TS
### Describe the solution you'd like

convert @woocommerce/csv-export to TypeScript and output type definitions

### Describe alternatives you've considered

_No response_

### Additional context

Part of https://github.com/woocommerce/woocommerce/issues/32179

### PR DESCRIPTION
**Pull request by chihsuan**
### All Submissions:

-   [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md)?
-   [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
-   [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

Closes #33102.

This PR migrates @woocommerce/number to TS and updates the JSDoc types

### How to test the changes in this Pull Request:

1. Run `pnpm nx build @woocommerce/csv-export`
2. Observe that packages/js/csv-export/build-types is generated
3. Should run `pnpm nx test @woocommerce/csv-export` without errors

### Other information:

-   [x] Have you added an explanation of what your changes do and why you'd like us to include them?
-   [ ] Have you written new tests for your changes, as applicable?
-   [x] Have you successfully run tests with your changes locally?
-   [x] Have you created a changelog file for each project being changed, ie `pnpm nx changelog <project>`?

### FOR PR REVIEWER ONLY:

-   [ ] I have reviewed that everything is sanitized/escaped appropriately for any SQL or XSS injection possibilities. I made sure Linting is not ignored or disabled.

### CODE REVIEW

File: packages/js/csv-export/src/index.ts [hunk 1]
```diff
@@ -2,9 +2,21 @@
...
  */
 import moment from 'moment';
-import { saveAs } from 'browser-filesaver';
+import { saveAs } from 'browser-filesaver'; // TODO: Replace this with https://www.npmjs.com/package/file-saver since browser-filesaver is not maintained anymore.
 
-function escapeCSVValue( value ) {
+export type Header = {
+	label: string;
+	key: string;
+};
```
Comment (chihsuan) [inline packages/js/csv-export/src/index.ts:10]: Ref:
https://github.com/woocommerce/woocommerce/blob/16a3fa7b38f36759e6937fe0ddb8d2a224ef773f/packages/js/csv-export/src/__mocks__/mock-headers.js#L1-L42

File: packages/js/csv-export/src/index.ts [hunk 2]
```diff
@@ -2,9 +2,21 @@
...
+
+export type RowItem = {
+	display: string;
+	value: string | number;
+};
```
Comment (chihsuan) [inline packages/js/csv-export/src/index.ts:15]: Ref: 

https://github.com/woocommerce/woocommerce/blob/16a3fa7b38f36759e6937fe0ddb8d2a224ef773f/packages/js/csv-export/src/__mocks__/mock-rows.js#L1-L44

File: packages/js/csv-export/typings/index.d.ts [hunk 3]
```diff
@@ -0,0 +1,5 @@
+declare module 'browser-filesaver' {
+	declare function FileSaver(data: Blob | string, filename?: string, options?: FileSaver.FileSaverOptions): void;
```
Comment (chihsuan) [inline packages/js/csv-export/typings/index.d.ts]: ref: https://github.com/tmpvar/browser-filesaver#syntax

**Comment by botwoo** [comment]
:bar_chart: Test reports for this pull request have been published and are accessible through the following links:

- [REST API Test Report](https://woocommerce.github.io/woocommerce-test-reports/pr/33131/api/)
- E2E Test Report (TBC)

Latest commit referenced in the reports: `Update filesaver type` 2c69e784607b5df0a907108a9f541fbc6a8424be
This comment will automatically be updated with the latest referenced commit when you push new changes to this pull request.

---

_Visit the [WooCommerce Test Reports](https://woocommerce.github.io/woocommerce-test-reports) homepage to view all published reports. See the [FAQs](https://github.com/woocommerce/woocommerce-test-reports/wiki/FAQs) page if you're having problems accessing them._

**Review by rjchow — APPROVED** [review]
Nice work hunting down the formats! Thanks!

**Pull request merged**